pandas/core/series.py

```diff
@@ -2561,27 +2561,44 @@ def memory_usage(self, index=True, deep=False):
 
     def take(self, indices, axis=0, convert=True, is_copy=False, **kwargs):
         """
-        return Series corresponding to requested indices
+        Return a Series corresponding to requested indices.
 
         Parameters
         ----------
-        indices : list / array of ints
-        convert : translate negative to positive indices (default)
+        indices : array-like
+            An array of ints indicating which positions to take.
+        axis : int, default 0
+            The axis on which to select elements. This implementation
+            ignores this parameter because there is only one axis.
+        convert : bool, default True
+            .. deprecated:: 0.21.0
+
+            Whether to translate negative to positive indices. In the future,
+            this function will always translate negative to positive indices.
+
+        is_copy : bool, default False
+            Whether to return a copy of the original object or not. This
+            implementation ignores this parameter.
 
         Returns
         -------
         taken : Series
+            An array-like containing the elements taken from the object.
 
-        See also
+        See Also
         --------
         numpy.ndarray.take
+        numpy.take
         """
+
         if kwargs:
             nv.validate_take(tuple(), kwargs)
 
-        # check/convert indicies here
-        if convert:
-            indices = maybe_convert_indices(indices, len(self._get_axis(axis)))
```

Comment:
Not sure this is safe to remove - with non numpy types, we call to an internal take impl that doesn't bounds check.  Can you try `pd.Series([1,2,3], dtype='category').take([-1, -2])` on your branch?

Reply:
Okay, so that breaks:

~~~python
2    NaN
1    1.0
dtype: category
Categories (3, int64): [1, 2, 3]
~~~

That being said, that call works without `dtype='category'`, meaning we *do* handle negative indices correctly in some cases already.  I might as well add that handling for `category` too.